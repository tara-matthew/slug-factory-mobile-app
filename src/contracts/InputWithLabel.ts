export enum TitleSize {
    Large = "large",
    Medium = "medium",
}

export interface IInputWithLabelProps {
    title: string;
    value?: string;
    titleSize: TitleSize;
    inputProps?: object;
    placeholder: string;
    isMultiline: boolean;
    sendDataToParent: (placeholder: string, text: string) => void;

}
