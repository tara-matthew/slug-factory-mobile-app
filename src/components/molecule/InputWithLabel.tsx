import React from "react";
import { Text, TextInput, View } from "react-native";
import {IInputWithLabelProps, TitleSize} from "../../contracts/InputWithLabel";

const InputWithLabel = ({ title, titleSize, value = null, inputProps = {}, placeholder, isMultiline, sendDataToParent }: IInputWithLabelProps) => {
    const inputStyling = isMultiline
        ? {
                ...inputProps,
                style: { height: 50 },
                editable: true,
                multiline: true,
                numberOfLines: 10,
            }
        : inputProps;
    const titleClassName = titleSize === TitleSize.Large ? "text-lg" : "text-md";

    return (
        <View>
            <Text className={ `font-bold ${titleClassName}` }>{title}</Text>
            <View className="bg-black/5 w-full p-5 rounded-2xl mb-7">
                <TextInput
                    { ...inputStyling }
                    value={ value }
                    placeholder={ placeholder }
                    onChangeText={ text => sendDataToParent(placeholder, text) }
                />
            </View>
        </View>

    );
};

export default InputWithLabel;
