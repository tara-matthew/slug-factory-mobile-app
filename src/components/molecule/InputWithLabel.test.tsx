import React from "react";
import { render } from "@testing-library/react-native";
import InputWithLabel from "./InputWithLabel";

it("renders correctly", () => {
    const mockSendDataToParent = jest.fn();
    const { toJSON } = render(<InputWithLabel isMultiline placeholder="placeholder" sendDataToParent={ mockSendDataToParent } title="title" />);

    expect(toJSON()).toMatchSnapshot();
});

it.todo("renders the correct title class");
it.todo("renders the correct styling depending on whether the input is multiline");
