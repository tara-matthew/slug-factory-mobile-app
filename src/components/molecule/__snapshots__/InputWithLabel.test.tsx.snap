// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    className="font-bold text-md"
  >
    title
  </Text>
  <View
    className="bg-black/5 w-full p-5 rounded-2xl mb-7"
  >
    <TextInput
      defaultValue={null}
      editable={true}
      multiline={true}
      numberOfLines={10}
      onChangeText={[Function]}
      placeholder="placeholder"
      style={
        {
          "height": 60,
        }
      }
      value={null}
    />
  </View>
</View>
`;
